euclidigons.lua

```diff
@@ -621,7 +737,7 @@ function enc(n, d)
 				edit_shape.rate = edit_shape.rate + d * 0.0015
 			elseif held_keys[3] then
 				-- set note
-				edit_shape.edits.note = edit_shape.edits.note + d
+				edit_shape.edits.note = util.clamp(edit_shape.edits.note + d, -64, 73)
```

Comment:
This isn't perfect, in face of more limited scales there are fewer possible notes. But it's better than nothing because otherwise it was quite easy to edit the note to a ridiculously high or low value. Then you had to dial a long way back to see any change. With this clamp you're never really far away.